test/namespaced_dependency/CMakeLists.txt

```diff
@@ -8,6 +8,16 @@ project(
 
 set(PROJECT_NAMESPACE "ns")
 add_library(${PROJECT_NAME} STATIC source/namespaced_dependency.cpp)
+
+if(${CMAKE_VERSION} VERSION_GREATER_EQUAL "3.23.0")
+  # required in CMake 3.23 and more
+  set(CMAKE_VERIFY_INTERFACE_HEADER_SETS ON)
+  target_sources(
+    ${PROJECT_NAME} PUBLIC FILE_SET HEADERS BASE_DIRS include FILES
+                           include/namespaced_dependency/namespaced_dependency.h
+  )
+endif()
```

Comment:
The Problem here is that this feature is only available starting with` CMake v3.23.0`

Reply:
Then we should have a separate test that fails with older versions of CMake. Otherwise it's difficult to tell which part of the tests have run and which not.

Reply:
@TheLartians do you now all this new features in `CMake` v3.24?

[dependency-providers](https://cmake.org/cmake/help/latest/guide/using-dependencies/index.html?highlight=fetchcontents#dependency-providers)

Reply:
I haven't had time to take a deep dive, but you can find a discussion in the CPM repository as well: https://github.com/cpm-cmake/CPM.cmake/discussions/362